src/utils/EncodingUtils.ts

```diff
@@ -0,0 +1,43 @@
+import ThundraLogger from '../ThundraLogger';
+import zlib from 'zlib';
+
+export const ENCODING_FUNCTIONS: { [key: string]: any } = {
+    br: zlib.brotliDecompressSync,
+    brotli: zlib.brotliDecompressSync,
+    gzip: zlib.gunzipSync,
+    deflate: zlib.deflateSync,
+};
+
+class EncodingUtils {
+
+    private constructor() {
+    }
+
+    static getJsonPayload(data: any, encoding: string) {
+        try {
+            let decodedData = data;
+            if (ENCODING_FUNCTIONS[encoding]) {
+                try {
+                    decodedData = ENCODING_FUNCTIONS[encoding](data);
+                } catch (err) {
+                    ThundraLogger.debug(`<EncodingUtils> Could decode data with ${encoding}`, err.message);
+                }
+            }
+
+            const decodedDataRef = decodedData;
+            try {
+                /**
+                 * Make sure data is json
+                 */
+                JSON.parse(decodedDataRef);
```

Comment:
The response doesn't need to be json. It can also be xml or any other format and we are able to show non-json responses in our UI. So we should just decode if it is encoded with supported formats. Or just return `toString` of given data. 

Also another reason for removing redundant (because it is not required to have json formatted response) `JSON.parse(...)` here is that it add extra performance overhead by parsing data for most of the cases.

So the method name should be `getPayload` and `JSON.parse(...)` should be removed here.